### PR TITLE
get longest palindrome

### PR DESCRIPTION
**Pull request by changweidalaifu6**
Updated code and added comments

### CODE REVIEW

**Comment by vil02** [comment]
Why did you close #703?

**Comment by changweidalaifu6** [comment]
#703 It's not clear why the other tests didn't pass, I didn't change any of the other code.

**Comment by vil02** [comment]
> #703 It's not clear why the other tests didn't pass, I didn't change any of the other code.

It is just some _random failure_ - some tests in this repository are not deterministic. In general - this is not a big deal.

**Comment by changweidalaifu6** [comment]
I thought it was my problem.....

**Comment by vil02** [comment]
In any case, the question about `aabb` from #703 remains valid.

**Comment by changweidalaifu6** [comment]
aabb theoretically outputs either aa or bb, here it only outputs bb

**Comment by changweidalaifu6** [comment]
I think that should be allowed.

**Comment by vil02** [comment]
The code looks quite good. Please update the _doc string_ by adding some explanation how do you find the longest palindrome, adding a sentence about _non-unique_ solution (I guess, your function returns _the one most on the right_).

Also, please add a test case with `"aabb"` (with a suitable comment).

**Comment by changweidalaifu6** [comment]
Is this build error due to the machine?  
Added comments to the left and right windows of the dp here, I think that should make it more palatable

**Comment by changweidalaifu6** [comment]
This compilation error is a machine problem？

File: src/dynamic_programming/get_longest_palindrome.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+pub fn get_longest_palindrome(s: &str) -> String {
+    if s.is_empty() {
+        return "".to_string();
+    }
+    let s: Vec<char> = s.chars().collect();
+    let n = s.len();
+    // dp indicates whether it is a palindrome
+    let mut dp = vec![vec![true; n]; n];
+    // res record the indexes before and after the palindrome
```
[the remark this answers is not in the document]
Comment (changweidalaifu6) [reply src/dynamic_programming/get_longest_palindrome.rs:18]: res[0] is the index at the beginning of the palindrome，res[1] is the index of the end of the palindrome

File: src/dynamic_programming/get_longest_palindrome.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Output: "bdb"
+*/
+
+pub fn get_longest_palindrome(s: &str) -> String {
+    if s.is_empty() {
+        return "".to_string();
+    }
+    let s: Vec<char> = s.chars().collect();
+    let n = s.len();
+    // dp indicates whether it is a palindrome
```
[the remark this answers is not in the document]
Comment (changweidalaifu6) [reply src/dynamic_programming/get_longest_palindrome.rs:16]: It's a bit redundant here, dp is the string of the dynamic programming process

File: src/dynamic_programming/get_longest_palindrome.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                }
+            )*
+        }
+    }
+    test_get_longest_palindrome! {
+        empty_input: ("", ""),
+        basic_1: ("abdbc", "bdb"),
+        basic_2: ("abyxycbabcyxy", "yxycbabcyxy"),
+        // Theoretically it is possible to return either aa or bb,
+        // there are multiple palindromes of the same length, here only the rightmost one is returned
```
[the remark this answers is not in the document]
Comment (changweidalaifu6) [reply src/dynamic_programming/get_longest_palindrome.rs:62]: You have a point, I don't think it's necessary to get hung up on whether the longest is unique or not, 
it's possible for a tournament to have a double winner as well

**Comment by vil02** [comment]
@changweidalaifu6 after rethinking the whole thing, I am not sure anymore if this contribution is suitable for this repository. The problem is very specific (it is a _leetcode_ problem after all) and quite cumbersome to state properly (well, the function should be called something like `get_length_of_the_palindrome_with_max_length`).

**Comment by changweidalaifu6** [comment]
It's true that it's a bit buttoned up, but it's very tight

**Pull request closed without merging**